interfaces/machine_parameters/ring.py

```diff
@@ -0,0 +1,598 @@
+
+        # Getting all sections and checking their types
+        if not hasattr(RingSection_list, '__iter__'):
+            RingSection_list = (RingSection_list,)
+        if not all(isinstance(s, RingSection) for s in RingSection_list):
+            raise excpt.InputError(
+                "The RingSection_list should be exclusively composed " +
+                "of RingSection object instances.")
+
+        self.RingSection_list = RingSection_list
+        self.n_sections = len(self.RingSection_list)
+
+        # Extracting the length of sections to get circumference on design
+        # orbit
+        self.section_length_design = np.array(
+            [section.length_design for section in self.RingSection_list])
+        self.circumference_design = np.sum(self.section_length_design)
+
+        # Computing ring radius on design orbit
+        self.radius_design = self.circumference_design / (2 * np.pi)
+
+        # Extracting the bending radius from all sections
+        self.bending_radius = np.array([
+            section.bending_radius for section in self.RingSection_list])
+
+        # Extracting the synchronous data from the sections,
+        # converting to momentum
+        # and checking if the synchronous data type and sizes are correct
+        momentum_list = []
+        momentum_by_turn = True
+        for index_section, section in enumerate(self.RingSection_list):
+
+            momentum = section.synchronous_data.convert(
+                self.Particle.mass,
+                self.Particle.charge,
+                self.bending_radius[index_section],
+                inPlace=False)
+
+            momentum_list.append(momentum)
+
+            if index_section == 0:
+
+                if momentum.timebase == 'by_time':
+                    self.synchronous_data = momentum
+                    momentum_by_turn = False
+
+            else:
+
+                if momentum.timebase != momentum_list[0].timebase:
+
+                    raise excpt.InputError(
+                        'The synchronous data for' +
+                        'the different sections is mixing time and turn ' +
+                        'based programs which is not supported.')
+
+                if (momentum_list[0].timebase == 'by_time'):
+                    if (momentum == momentum_list[0]).all():
+
+                        warn_message = 'The synchronous data for all sections ' + \
+                            'are defined time based and ' + \
+                            'are identical. Presently, ' + \
+                            'the momentum is assumed constant for one turn over ' + \
+                            'all sections, no increment in delta_E from ' + \
+                            'one section to the next. Please use custom ' + \
+                            'turn based program if needed.'
+                        warnings.warn(warn_message)
+
+                    else:
+
+                        raise excpt.InputError(
```

Comment:
Perhaps a better exception would be `NotImplementedError`.  The input here is fine, we just haven't yet figured out how to treat it.
```suggestion
                        raise NotImplementedError(
```